### PR TITLE
Added possibility to switch off reusing the Node

### PR DESCRIPTION
**Pull request by statut**
Our project should run integration tests several times so that we need to reuse the ElasticSearch Node

### CODE REVIEW

File: src/main/java/com/github/alexcojocaru/mojo/elasticsearch/StartElasticsearchNodeMojo.java [hunk 1]
```diff
@@ -105,7 +110,7 @@ public void execute() throws MojoExecutionException
 
     protected ElasticsearchNode startNode(Settings settings) throws MojoExecutionException
     {
-        if (getNode() != null)
+        if (reuseElasticNode && getNode() != null)
```
Comment (alexcojocaru) [inline src/main/java/com/github/alexcojocaru/mojo/elasticsearch/StartElasticsearchNodeMojo.java]: Can you describe the use case you are trying to solve with this?

File: src/main/java/com/github/alexcojocaru/mojo/elasticsearch/StartElasticsearchNodeMojo.java [hunk 2]
```diff
@@ -54,6 +54,11 @@
     protected boolean autoCreateIndex;
 
     /**
+     * @parameter default-value=true
+     */
+    protected boolean reuseElasticNode = true;
```
Comment (alexcojocaru) [inline src/main/java/com/github/alexcojocaru/mojo/elasticsearch/StartElasticsearchNodeMojo.java]: Please add documentation and test(s).

File: src/main/java/com/github/alexcojocaru/mojo/elasticsearch/StartElasticsearchNodeMojo.java (hunk 1 above)
Comment (alexcojocaru) [inline src/main/java/com/github/alexcojocaru/mojo/elasticsearch/StartElasticsearchNodeMojo.java]: My bad for not reading the description you added to the merge request; I understand now why you need this parameter.

If reuseElasticNode is false, then the node property is reset and you loose the reference to the previously started ES instance.

**Comment by statut** [comment]
I reverted my changes since there was no problem in reusing. Instead of this I added checking: getNode().isClose() before returning node in startNode() method and wrote test that reproduces bug: if we stopped the node and then calls startNode(), startNode() method returns the node from the context but that node is invalid.

**Comment by alexcojocaru** [comment]
Looks good; thanks.

**Comment by statut** [comment]
Could you, please, tell me, when these changes will be in the maven central? Thanks

**Comment by alexcojocaru** [comment]
I am releasing a new version (2.3) of the plugin, it will be available in maven central in a couple of hours.

**Comment by statut** [comment]
Бонус
Ют яЯ**":"'

сб, 19 нояб. 2016 г., 8:14 alexcojocaru <notifications@github.com>:

**Comment by statut** [comment]
Thank you very much!

**Pull request merged**